cg/utils/files.py

```diff
@@ -41,3 +42,40 @@ def rename_file(file_path: Path, renamed_file_path: Path) -> None:
 def is_pattern_in_file_path_name(file_path: Path, pattern: str) -> bool:
     """Check if a pattern is in a file path."""
     return pattern in file_path.name
+
+
+def get_directory_creation_time_stamp(directory_path: Path) -> float:
+    """
+    Return time stamp that a directory is created.
+    Raises:
+        FileNotFoundError if the directory does not exist.
+        ValueError if the specified path is not a directory.
+    """
+    if not directory_path.exists():
+        raise FileNotFoundError(f"Directory with path {directory_path} is not found.")
+
+    if not directory_path.is_dir():
+        raise ValueError(f"Specified path {directory_path} is not a directory.")
+
+    return os.stat(directory_path).st_ctime
+
+
+def remove_directory_and_contents(directory_path):
+    """
+    Delete a directory and its contents.
+    Excepts OSError.
+    """
+    try:
+        shutil.rmtree(directory_path)
+        LOG.info(f"Successfully removed the directory and its contents: {directory_path}")
+    except OSError as error:
+        LOG.error(f"Failed to remove the directory {directory_path} and its contents: {error}")
```

Comment:
I changed the functionality to 'Except Exception' and `raise OsError` since the `shutil.rmtree()` function actually raises some other things as well potentially.